crates/noirc_frontend/src/monomorphization/ast.rs

```diff
@@ -319,9 +325,9 @@ impl std::fmt::Display for Type {
                 let elements = vecmap(elements, ToString::to_string);
                 write!(f, "({})", elements.join(", "))
             }
-            Type::Function(args, ret) => {
+            Type::Function(args, ret, env) => {
                 let args = vecmap(args, ToString::to_string);
-                write!(f, "fn({}) -> {}", args.join(", "), ret)
+                write!(f, "fn({}) -> {} [{}]", args.join(", "), ret, env)
```

Comment:
For consistency we should also probably have the same "with closure environment" printout here as well

Reply:
ok, fixed(in next commit)